modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -725,8 +725,11 @@ private CompletableFuture<?> updateAssignmentInternal(ConfigurationNotificationE
 
                 placementDriver.updateAssignment(replicaGrpId, newConfiguration.peers().stream().map(Peer::consistentId).collect(toList()));
 
-                PendingComparableValuesTracker<HybridTimestamp> safeTime = new PendingComparableValuesTracker<>(new HybridTimestamp(1, 0));
-                PendingComparableValuesTracker<Long> storageIndexTracker = new PendingComparableValuesTracker<>(0L);
+                var safeTimeTracker = new PendingComparableValuesTracker<>(new HybridTimestamp(1, 0));
+                var storageIndexTracker = new PendingComparableValuesTracker<>(0L);
+
+                ((InternalTableImpl) internalTbl).updatePartitionSafeTimeTracker(partId, safeTimeTracker);
```

Comment:
Why is this cast necessary? Isn't it possible to add the methods to the interface?

Reply:
I follow the example with `RaftGroupService`, getting this is a "public" api `org.apache.ignite.internal.table.InternalTable#partitionRaftGroupService`, updating in the implementation `org.apache.ignite.internal.table.distributed.storage.InternalTableImpl#updateInternalTableRaftGroupService`.

I think that for now let it remain as it is.

Reply:
Ok